pkg/scheduler/algorithm/priorities/interpod_affinity.go

```diff
@@ -135,6 +135,9 @@ func (ipa *InterPodAffinity) CalculateInterPodAffinityPriority(pod *v1.Pod, node
 	var maxCount, minCount int64
 
 	processPod := func(existingPod *v1.Pod) error {
+		if pm.firstError != nil {
```

Comment:
This is an unnecessary optimization that could rarely be triggered rarely and even if it is triggered, the saving is minimal as we only skip processing remaining pods on one node.